src/components/OpenPopup.jsx

```diff
@@ -0,0 +1,32 @@
+import React, {Component} from "react";
+import Popup from "./Popup";
+
+//this class is used to open the popup activity
+//It contains the functions necessary for opening and closing the popup
+
+class OpenPopup extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            showPopup: false
+        };
+    }
+    togglePopup() {
+        this.setState({
+            showPopup: !this.state.showPopup
+        });
+    }
+    render() {
+        return (
+            <div className="App">
+                <button onClick={this.togglePopup.bind(this)}>show popup</button>
```

Comment:
Please avoid using `.bind(this)` in render method. If you don't want to use arrow function, bind the methods in contructor like :

```js
  this.togglePopup = this.togglePopup.bind(this)
```